src/Microsoft.SqlTools.ServiceLayer/Connection/ListDatabaseRequestHandler.cs

```diff
@@ -57,43 +61,66 @@ abstract class ListDatabaseRequestHandler<T> : IListDatabaseRequestHandler
     {
         private static readonly string[] SystemDatabases = new string[] { "master", "model", "msdb", "tempdb" };
 
-        public abstract string QueryText { get; }
-
         public ListDatabasesResponse HandleRequest(ISqlConnectionFactory connectionFactory, ConnectionInfo connectionInfo)
         {
             ConnectionDetails connectionDetails = connectionInfo.ConnectionDetails.Clone();
+            var serverConnection = ConnectionService.OpenServerConnection(connectionInfo);
+
+            SMO.Server SQLServer = new SMO.Server(serverConnection);
+
 
-            // Connect to master
-            connectionDetails.DatabaseName = "master";
-            using (var connection = connectionFactory.CreateSqlConnection(ConnectionService.BuildConnectionString(connectionDetails), connectionDetails.AzureAccountToken))
+            List<T> result = new List<T>();
+            ListDatabasesResponse response = new ListDatabasesResponse();
+
+
+            foreach (SMO.Database db in SQLServer.Databases)
```

Comment:
I have a concern about the performance here, with this change, you have changed the required server roundtrips from 1 to 1+x (number of dbs). that is what i tried to avoid when i was working on this.